server/homebrew.api.js

```diff
@@ -287,15 +287,21 @@ const api = {
 
 		res.status(200).send(saved);
 	},
+	getGoogleAuth : function(account, res){
+		const auth = GoogleActions.authCheck(account, res);
+		return auth;
+	},
 	deleteGoogleBrew : async (account, id, editId, res)=>{
-		const auth = await GoogleActions.authCheck(account, res);
+		const auth = api.getGoogleAuth(account, res);
 		await GoogleActions.deleteGoogleBrew(auth, id, editId);
 		return true;
 	},
 	deleteBrew : async (req, res, next)=>{
 		// Delete an orphaned stub if its Google brew doesn't exist
 		try {
-			await api.getBrew('edit')(req, res, ()=>{});
+			const { googleId } = api.getId(req);
+			const auth = googleId ? api.getGoogleAuth(req.account, res) : undefined;
+			await api.getBrew('edit', false, auth)(req, res, ()=>{});
```

Comment:
Can you confirm this allows the "restricted sharing" google brew to be deleted?

Is this going to cause issues with co-authors who want to delete the brew from their own account in normal cases? i.e., if User A makes the Google Brew, invites User B as an author, then if User B tries to delete, my guess is it will fail because the Service Account has delete permissions but User B's account doesn't.

Reply:
1. Yes, it does - the document is trashed from the Google account to the Bin.
2. If I understand the scenario, Author B shouldn't be deleting the brew, but rather only removing themselves from the list of authors - which from memory is held in the MongoDB stub. If the owner of the Google Drive file attempts to delete it but there are other authors, the brew is moved to MongoDB and the original owner removed from the authors list.

Reply:
2) Yes, I mean removing themselves as an author. But with Author B no longer using ServiceAccount, isn't `getBrew()` going to fail for them and exit `deleteBrew()` prematurely before they are removed as an author later on in this function?

Reply:
@G-Ambatte Can you confirm that for a brew with two co-authors, if I delete a brew as Author B (non-owner), it just removes from my account and does not delete the original (or cause an error because it can't retrieve the brew without serviceauth)?

Reply:
> I think deleteBrew() will not work as expected for Author B.

I've been going over this again, and I think you're exactly right - if two authors both are linked to Google, the delete request will fail because of incorrect permissions, which Google normally returns as a 404 - there is no externally identifiable difference between "blocked because it doesn't exist" and "blocked because disallowed". This will trigger the `err.HBErrorCode == '02'` block and delete the whole stub.

I also realize now that this is because that is the intended purpose of this section of code; I was modifying the wrong section of code for my intended purpose.